kythe/java/com/google/devtools/kythe/util/schema/Nodes.java

```diff
@@ -120,9 +129,67 @@ public static Node normalizeNode(Node n) {
                 b.addFact(f);
               }
             });
+  }
+
+  /**
+   * Returns a {@link Node} from the given set of {@link Entry} protos. All {@link Entry}s must
+   * share the same source {@link VName}.
+   */
+  public static Node fromEntries(Iterable<Entry> entries) {
+    List<Fact> facts = new ArrayList<>();
+    List<Edge> edges = new ArrayList<>();
+    for (Entry e : entries) {
+      if (e.hasFact()) {
+        facts.add(e.getFact());
+      } else {
+        edges.add(e.getEdge());
```

Comment:
can't an entry contain both facts and edges?

Reply:
No, it's a `oneof`